src/soc/ibm/power9/istep_14_3.c

```diff
@@ -0,0 +1,36 @@
+#include <cpu/power/istep_14.h>
+
+static void p9_pcie_config(void)
+{
+	for(size_t PECIndex = 0; PECIndex < PEC_PER_PROC; ++PECIndex)
+	{
+		scom_and_for_chiplet(
+			pec_ids[PECIndex], P9N2_PEC_ADDREXTMASK_REG,
+			~PPC_BITMASK(0, 6));
+		scom_or_for_chiplet(
+			pec_ids[PECIndex], PEC_PBCQHWCFG_REG,
+			PPC_BIT(3) | PPC_BIT(7) | PPC_BIT(11)
+			|	PPC_BIT(PEC_PBCQHWCFG_REG_PE_DISABLE_OOO_MODE)
+			| PPC_BIT(PEC_PBCQHWCFG_REG_PE_CHANNEL_STREAMING_EN)
+			| PPC_BIT(PEC_PBCQHWCFG_REG_PE_DISABLE_WR_VG)
+			| PPC_BIT(PEC_PBCQHWCFG_REG_PE_DISABLE_WR_SCOPE_GROUP)
+			| PPC_BIT(PEC_PBCQHWCFG_REG_PE_DISABLE_INTWR_VG)
+			| PPC_BIT(PEC_PBCQHWCFG_REG_PE_DISABLE_INTWR_SCOPE_GROUP)
+			| PPC_BIT(PEC_PBCQHWCFG_REG_PE_DISABLE_RD_VG)
+			| PPC_BIT(PEC_PBCQHWCFG_REG_PE_DISABLE_RD_SCOPE_GROUP)
+			| PPC_BIT(PEC_PBCQHWCFG_REG_PE_DISABLE_TCE_SCOPE_GROUP)
+			| PPC_BIT(PEC_PBCQHWCFG_REG_PE_DISABLE_TCE_VG));
+		scom_or_for_chiplet(
+			pec_ids[PECIndex], PEC_NESTTRC_REG,
+			PPC_BIT(0) | PPC_BIT(3));
+		write_scom_for_chiplet(
+			pec_ids[PECIndex], PEC_PBAIBHWCFG_REG,
+			0xe00000 | PPC_BIT(PEC_PBAIBHWCFG_REG_PE_PCIE_CLK_TRACE_EN));
```

Comment:
See https://github.com/3mdeb/openpower-coreboot-docs/blob/scom_dumps/logs/scom_dumps/istep_14_3_proc_pcie_config.log. Only the last SCOM uses PEC ID, the rest uses `N2_CHIPLET_ID`. Those under Nest should also have `+ 0x400 * PECIndex` I think.

This covers lines up to 25 in the log linked above, what about the rest?

Reply:
About the missing part of the log, I have probably messed up, while trying to understand hostboot code, and removed part of it thinking it is never executed. I'll have look at it for the second time.

Reply:
Ok, for now it doesn't work anyway, probably due to missing SCOM init for that chiplet. I'll merge it as it is so I can start cleaning my code on top of it, commenting out call to `istep_14_3()`.